internal/templates/assisted-installer/template.go

```diff
@@ -0,0 +1,215 @@
+  annotations:
+    metaclusterinstall.openshift.io/sync-wave: "2"
+  name: "{{ .Site.ClusterName }}"
+  namespace: "{{ .Site.ClusterName }}"
+spec:
+  clusterName: "{{ .Site.ClusterName }}"
+  clusterNamespace: "{{ .Site.ClusterName }}"
+  clusterLabels:
+    cloud: auto-detect
+    vendor: auto-detect
+  applicationManager:
+    enabled: false
+  certPolicyController:
+    enabled: false
+  iamPolicyController:
+    enabled: false
+  policyController:
+    enabled: true
+  searchCollector:
+    enabled: false`
+
+const NMStateConfig = `apiVersion: agent-install.openshift.io/v1beta1
+kind: NMStateConfig
+metadata:
+  annotations:
+    metaclusterinstall.openshift.io/sync-wave: "1"
+  name: "{{ .SpecialVars.CurrentNode.HostName }}"
+  namespace: "{{ .Site.ClusterName }}"
+  labels:
+    nmstate-label: "{{ .Site.ClusterName }}"
+spec:
+  config:
+{{ .SpecialVars.CurrentNode.NodeNetwork.NetConfig | toYaml | indent 4}}
+  interfaces:
+{{ .SpecialVars.CurrentNode.NodeNetwork.Interfaces | toYaml | indent 4 }}`
+
+const ManagedCluster = `apiVersion: cluster.open-cluster-management.io/v1
+kind: ManagedCluster
+metadata:
+  name: "{{ .Site.ClusterName }}"
+  labels:
+{{ .Site.ClusterLabels | toYaml | indent 4 }}
+  annotations:
+    metaclusterinstall.openshift.io/sync-wave: "2"
+spec:
+  hubAcceptsClient: true`
+
+const BareMetalHost = `apiVersion: metal3.io/v1alpha1
+kind: BareMetalHost
+metadata:
+  name: "{{ .SpecialVars.CurrentNode.HostName }}"
+  namespace: "{{ .Site.ClusterName }}"
+  annotations:
+    metaclusterinstall.openshift.io/sync-wave: "1"
+    inspect.metal3.io: "{{ .SpecialVars.CurrentNode.IronicInspect }}"
+{{ if .SpecialVars.CurrentNode.NodeLabels }}
+    bmac.agent-install.openshift.io.node-label:
+{{ .SpecialVars.CurrentNode.NodeLabels | toYaml | indent 6 }}
+{{ end }}
+    bmac.agent-install.openshift.io/hostname: "{{ .SpecialVars.CurrentNode.HostName }}"
+{{ if .SpecialVars.CurrentNode.InstallerArgs  }}
+    bmac.agent-install.openshift.io/installer-args: {{ .SpecialVars.CurrentNode.InstallerArgs  }}
+{{ end }}
+{{ if .SpecialVars.CurrentNode.IgnitionConfigOverride }}
+    bmac.agent-install.openshift.io/ignition-config-overrides: {{ .SpecialVars.CurrentNode.IgnitionConfigOverride }}
+{{ end }}
+    bmac.agent-install.openshift.io/role: "{{ .SpecialVars.CurrentNode.Role }}"
+  labels:
+    infraenvs.agent-install.openshift.io: "{{ .Site.ClusterName }}"
+spec:
+  bootMode: "{{ .SpecialVars.CurrentNode.BootMode }}"
+  bmc:
+    address: "{{ .SpecialVars.CurrentNode.BmcAddress }}"
+    disableCertificateVerification: true
+    credentialsName: "{{ .SpecialVars.CurrentNode.BmcCredentialsName.Name }}"
+  bootMACAddress: "{{ .SpecialVars.CurrentNode.BootMACAddress }}"
+  automatedCleaningMode: "{{ .SpecialVars.CurrentNode.AutomatedCleaningMode }}"
+  online: true
+{{ if (anyFieldDefined .SpecialVars.CurrentNode.RootDeviceHints) }}
+  rootDeviceHints:
+{{ .SpecialVars.CurrentNode.RootDeviceHints | toYaml | indent 4 }}
+{{ end }}`
+
```

Comment:
Original siteConfig had the userData field. Is this no longer supported?

Reply:
It was removed. Discussion around the `userData` field: https://github.com/sakhoury/siteconfig/pull/1#discussion_r1561578977.